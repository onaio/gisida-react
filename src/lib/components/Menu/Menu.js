import React, { Component } from 'react';
import { connect } from 'react-redux'
import PropTypes from 'prop-types';
import { Actions } from 'gisida';
import Layers from '../Layers/Layers';
import './Menu.scss';

const mapStateToProps = (state, ownProps) => {
  const MAP = state[ownProps.mapId] || { layers: {} };
  const { LAYERS } = state;
  let categories;
  // let layers;

  if (Object.keys(LAYERS.groups).length) {
    // build list of LAYERS.categories populated with layers from MAP.layers 
    categories = Object.keys(LAYERS.groups).map((group) => {
      return {
        category: group,
        layers: LAYERS.groups[group].map((l) => MAP.layers[l])
          .filter((l) => typeof l !== 'undefined'),
      };
    });
  } else if (Object.keys(MAP.layers).length) {
    categories = {};
    let category;

    Object.keys(MAP.layers).forEach((l) => {
      if (MAP.layers[l].category) {
        category = MAP.layers[l].category;
        if (!categories[category]) {
          categories[category] = {
            category,
            layers: [],
          };
        }
        categories[category].layers.push(MAP.layers[l]);
      }
    });

    categories = Object.keys(categories).map(c => categories[c]);
  }

  // todo - support layers without categories
  // if (!Object.keys(categories).length) {
  //   categories = null;
  //   layers = Object.keys(MAP.layers).map(l => MAP.layers[l]);
  // }

  // Get current region
  const currentRegion = state.REGIONS && state.REGIONS.length ?
    state.REGIONS.filter(region => region.current)[0].name : '';

  return {
    categories,
    // layers, // todo - support layers without categories
    LAYERS,
    menuId: 'sector-menu-1',
    mapTargetId: '',
    regions: state.REGIONS,
    currentRegion: currentRegion,
    loaded: state.APP.loaded,
    preparedLayers: MAP.layers,
  };
}

class Menu extends Component {
  constructor(props) {
    super(props)
    this.state = {
      openCategories: [],
      openMenu: true
    }
  }

  onToggleMenu = (e) => {
    e.preventDefault();
    this.setState({ openMenu: !this.state.openMenu });
  }

  onCategoryClick = (e, category) => {
    e.preventDefault();

    const openCategories = this.state.openCategories;
    const index = openCategories.indexOf(category);

    if (index > -1) {
      openCategories.splice(openCategories.indexOf(category), 1);
    } else openCategories.push(category);
    this.setState({ openCategories });
  }

  onRegionClick = (e) => {
    const region = e.target.value;
    this.props.dispatch(Actions.changeRegion(region));
  }

  render() {
    const mapId = this.props.mapId;
    const categories = this.props.categories;
    let sortedCategories;
    if (categories && categories.length > 0) {
      sortedCategories = categories.sort((a, b) =>
        a.category.localeCompare(b.category)
      );
    }
    const regions = this.props.regions;
    const currentRegion = this.props.currentRegion;
    const preparedLayers = this.props.preparedLayers;
    return (
      <div>
        {this.props.children ? this.props.children :
          <div>
            {this.props.loaded ?
              // Menu Wrapper
              <div id={`${mapId}-menu-wrapper`} className="menu-wrapper">
                {/* Open button menu */}
                <a onClick={e => this.onToggleMenu(e)} className="open-btn"
                  style={{ display: this.state.openMenu ? 'none' : 'block' }}>
                  <span className="glyphicon glyphicon-menu-hamburger"></span>
                </a>
                {/* Menu */}
                <div id={`${mapId}-menu`} className="sectors-menu"
                  style={{ display: this.state.openMenu ? 'block' : 'none' }}>
                  {/* Close menu button */}
                  <a className="close-btn" onClick={e => this.onToggleMenu(e)}>
                    <span className="glyphicon glyphicon-remove"></span>
                  </a>
                  {/* Menu List*/}
                  <ul className="sectors">
                    {regions && regions.length ?
                      <li className="sector">
                        <a onClick={e => this.onCategoryClick(e, 'Regions')}>Regions
                          <span className="caret" />
                        </a>
                        <ul className="layers">
                          {regions && regions.length ?
                            regions.map((region, i) =>
                              (<li className={`region ${mapId}`} key={region.name}>
                                <input
                                  id={region.name}
                                  key={region.name}
                                  name="region"
                                  type="radio"
                                  value={region.name}
                                  checked={!!region.current}
                                  onChange={e => this.onRegionClick(e)}
                                />
                                <label htmlFor={region.name}>{region.name}</label>
                              </li>)) :
                            <li></li>
                          }
                        </ul>
                      </li> : <li />}
                    {(categories && categories.length) > 0 ?
                      sortedCategories.map((category, i) =>
                        (<li className="sector" key={i}>
                          <a onClick={e => this.onCategoryClick(e, category.category)}>{category.category}
                            <span
                              className={"category glyphicon " +
                                (this.state.openCategories.includes(category.category) ?
                                  "glyphicon-chevron-down" : "glyphicon-chevron-right")}
                            />
                          </a>
                          {
                            this.state.openCategories.includes(category.category) ?
                              <Layers
                                mapId={mapId}
                                layers={category.layers}
                                currentRegion={currentRegion}
                                preparedLayers={preparedLayers}
                              />
                              : <ul />}
                        </li>)) :
                      <li></li>
                    }
                  </ul>
                </div>
              </div> : ''}
          </div>
        }
      </div>
    );
  }
}

Menu.propTypes = {
  menuId: PropTypes.string.isRequired,
  // mapTargetId: PropTypes.string.isRequired,
  categories: PropTypes.arrayOf(PropTypes.any).isRequired,
};

export default connect(mapStateToProps)(Menu);
