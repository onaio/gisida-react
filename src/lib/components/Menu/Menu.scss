.menu-wrapper {
  position: fixed;
  z-index: 1000;
  overflow-y: auto;
}
.sectors-menu {
  width: 300px;
  color: #555;
  z-index: 1000;
  background: #f7f6ef;
  border-right: 1px solid #ddd;
  font-family: 'Montserrat', sans-serif;

  .close-btn {
    position: absolute;
    right: 15px;
    top: 15px;
    z-index: 1000;
    color: #555;
    display: block;
  }

  ul {
    list-style-type: none;
    margin-top: 0;
  }

  .sectors {
    padding: 0;
    li.sector {
      span.links {
        display: inline-flex;
        font-size: 18px;
        .hyperlink {
          color: #ccc;
          background-color: #f7f6ef;
          &:hover {
            color: #333;
          }
        }
        .description {
          display: inline-flex;
          top: 4px;
          right: 5px;
          font-size: 18px;
          text-align: right;
          color: #ccc;
          margin-top: 4px;
    
          span {
            cursor: pointer;
            top: 3px;
          }
    
          p {
            display: none;
            position: absolute;
            right: 1.5em;
            text-align: left;
            padding: 6px 10px;
            background-color: #333;
            margin-top: 31px;
            color: #eee;
            font-size: 11px;
            line-height: 1.5;
            width: 250px;
            z-index: 1;
            &:before {
              content: '';
              height: 0;
              width: 0;
              position: absolute;
              top: -11px;
              right: 20;
              border-left: 7px solid transparent;
              border-right: 7px solid transparent;
              border-bottom: 11px solid #333;
            }
          }
    
          &:hover {
            color: #333;
            p {
              display: block;
            }
          }
        }
      }
      span.sub-category-links {
        display: inline-flex;
        font-size: 18px;
        .hyperlink {
          color: #ccc;
          background-color: #f7f6ef;
          &:hover {
            color: #333;
          }
        }
        .description {
          display: inline-flex;
          top: 4px;
          right: 2.5px;
          font-size: 18px;
          text-align: right;
          color: #ccc;
          margin-top: 4px;
    
          span {
            cursor: pointer;
            top: 3px;
          }
    
          p {
            display: none;
            position: absolute;
            right: 1.5em;
            text-align: left;
            padding: 6px 10px;
            background-color: #333;
            margin-top: 31px;
            color: #eee;
            font-size: 11px;
            line-height: 1.5;
            width: 193px;
            z-index: 1;
    
            &:before {
              content: '';
              height: 0;
              width: 0;
              position: absolute;
              top: -11px;
              right: 20;
              border-left: 7px solid transparent;
              border-right: 7px solid transparent;
              border-bottom: 11px solid #333;
            }
          }
    
          &:hover {
            color: #333;
            p {
              display: block;
            }
          }
        }
      }
    }

    li.sector a {
      color: #555;
      display: block;
      font-weight: 600;
      padding: 25px 15px;
      background: rgba(255, 255, 255, 0.4);
      border-bottom: 1px solid #eee;
      text-decoration: none;
      &:hover,
      &:focus {
        text-decoration: none;
        color: #555;
      }
    }
    li.sector .hyperlink {
      padding: 7px 15px !important;
      border: none !important;
    }
  }

  .sectors li.search-sector a {
    color: #555;
    display: block;
    font-weight: 400;
    padding: 10px 10px;
    background: rgba(255, 255, 255, 0.4);
    border-bottom: 1px solid #eee;
    text-decoration: none;
    cursor: pointer;

    &:hover,
    &:focus {
      text-decoration: none;
      color: #555;
    }
  }

  .layers {
    padding: 0px 0 0 20px;

    li.layer {
      position: relative;
      display: block;
      width: 100%;
      padding-top: 10px;
    }

    li.region {
      margin-bottom: 5px;
      
      label {
        display: inline;
        margin: 0 0 0 5px;
      }
    }
    a.hyperlink {
        padding: 7px 15px !important;
        border: none !important;
    }
    a.sub-category {
      padding: 9px 20px !important;
      margin-right: 2px;
      margin-left: -16px;
      position: relative;
      font-weight: normal !important;

      .category {
        color: #777;
      }
    }
  }

  label {
    color: #555;
    font-weight: normal;
    font-size: 12px;
    width: 80%;
    display: block;
    margin: 0 0 0 20px;
  }

  input[type='checkbox'] {
    position: absolute;
    left: 0;
    top: 10px;
  }
}

.open-btn {
  position: fixed;
  color: #333;
  background: #fff;
  border: 2px solid #ddd;
  border-radius: 5px;
  width: 32px;
  line-height: 2;
  font-size: 15px;
  text-align: center;
  top: 10px;
}

#map-1-menu-wrapper .open-btn {
  left: 90px;
}

#map-2-menu-wrapper .open-btn {
  left: 53%;
}

#map-1-menu-wrapper {
  background: #f7f6ef;
  height: 100%;
  left: 80px;
}

#map-2-menu {
  height: 100%;
  top: 0;
  left: 52%;
  display: none;
  overflow-y: auto;
  position: fixed;
}

.category {
  margin-left: 5px;
  font-size: 80%;
}

.no-search-results {
  margin: 20px 4px;
  border-bottom: 1px solid #eee;
}
