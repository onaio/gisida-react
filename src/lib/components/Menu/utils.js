import memoize from 'memoize-one';

/**
 * Returns true if a user has permission to access layer, false otherwise
 * @param {string} layerId the layer name/id
 * @param {Object} authConfigs - Authentication configurations
 * @param {Object} userInfo - User details
 */
export const canAccessLayer = (layerId, authConfigs, userInfo) => {
  // list of users with access to the layer
  const users = authConfigs && authConfigs.LAYERS && authConfigs.LAYERS[layerId];

  return (
    (users && userInfo && users.includes(userInfo.username)) ||
    (authConfigs.LAYERS &&
      authConfigs.LAYERS.ALL &&
      authConfigs.LAYERS.ALL.includes(userInfo.username))
  );
};

/**
 * Return an accesible group layer. If the layer has no accessible children
 * return false, else return the accessible children
 * @param {Object} layer - Group layer
 * @param {Object} authConfigs - Authentication configurations
 * @param {Object} userInfo - Auth user details
 */
export const getAccessibleGroupLayer = (layer, authConfigs, userInfo) => {
  const accesibleGroup = {};

  Object.keys(layer).forEach(key => {
    const accessibleKeySubLayers = [];

    layer[key].layers.forEach(subLayer => {
      if (!subLayer.id) {
        const groupSubLayer = getAccessibleGroupLayer(subLayer, authConfigs, userInfo);

        if (groupSubLayer) {
          accessibleKeySubLayers.push(groupSubLayer);
        }
      } else {
        if (canAccessLayer(subLayer.id, authConfigs, userInfo)) {
          accessibleKeySubLayers.push(subLayer);
        }
      }
    });

    if (accessibleKeySubLayers.length > 0) {
      // Return sub layers which user has access to
      accesibleGroup[key] = {
        ...layer[key],
        layers: accessibleKeySubLayers,
      };
    }
  });

  // If user can access any layer with the group, return the group
  if (Object.keys(accesibleGroup).length > 0) {
    return accesibleGroup;
  }

  return false;
};

/**
 * Get which categories and their groups and nested groups user has
 * permission to view
 * @param {array} categories array of categories to filter
 * @param {Object} authConfigs - Authentication configurations
 * @param {Object} userInfo - Auth user details
 * @returns {array} Filtered categories
 */
export const getAccessibleCategories = memoize((categories, authConfigs, userInfo) => {
  if (!authConfigs && !userInfo) {
    // If no authenitcation, then all categories are accessible.
    return categories;
  }

  const filteredCategories = [];
  categories.forEach(category => {
    let accesibleLayers = [];

    category.layers.forEach(layer => {
      if (!authConfigs || !authConfigs.LAYERS) {
        // If auth exists but authconfigs have not loaded. Bug should be fixed from ONA data and gisida core
        accesibleLayers.push(layer);
      } else if (!layer.id) {
        const groupLayer = getAccessibleGroupLayer(layer, authConfigs, userInfo);

        if (groupLayer) {
          accesibleLayers.push(groupLayer);
        }
      } else {
        if (canAccessLayer(layer.id, authConfigs, userInfo)) {
          accesibleLayers.push(layer);
        }
      }
    });

    if (accesibleLayers.length > 0) {
      // Modify category layers with the new updated layers
      category.layers = accesibleLayers;
      filteredCategories.push(category);
    }
  });

  return filteredCategories;
});
