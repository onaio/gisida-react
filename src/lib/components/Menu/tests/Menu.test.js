import React from 'react';
import Menu from '../Menu.js';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { layer2 } from '../../../../../test/fixtures/defaultLayers.js';
import { categories } from './fixtures.js';
import { Actions } from 'gisida';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';

const initialState = {
  LAYERS: { groups: {} },
  APP: { loaded: true },
  CATEGORIES: categories,
  'map-1': {
    layers: { Regionboundaries: { ...layer2 } },
    openCategories: ['Boundaries'],
    primaryLayer: layer2.id,
    visibleLayerId: layer2.id,
  },
  REGIONS: [],
  VIEW: {
    showMap: false,
  },
};

const mockStore = configureStore();
const props = { mapId: 'map-1', menuId: 'menu-1' };
const groups = {
  Boundaries: [{ Regionboundaries: ['regionboundaries'] }],
};

describe('Menu Component', () => {
  it('renders without crashing', () => {
    const store = mockStore(initialState);
    shallow(<Menu {...props} store={store} />);
  });

  it('Menu component renders correctly with all props', () => {
    initialState.LAYERS.groups = { ...groups };
    initialState.REGIONS = [
      { name: 'reg1', current: false },
      { name: 'reg2', current: true },
    ];
    const store = mockStore(initialState);
    const wrapper = mount(<Menu {...props} store={store} />);
    // test snapshots
    expect(wrapper.find('Menu').props()).toMatchSnapshot('Menu props');
    expect(toJson(wrapper)).toMatchSnapshot();

    // test onToggleMenu
    const toggleMenuAction = jest.spyOn(Actions, 'toggleMenu');
    wrapper.find('a.open-btn').simulate('click');
    expect(toggleMenuAction).toHaveBeenCalledWith(props.mapId);
    wrapper.find('a.close-btn').simulate('click');
    expect(toggleMenuAction).toHaveBeenCalledTimes(2);

    // test onCategoryClick
    const toggleCategoriesAction = jest.spyOn(Actions, 'toggleCategories');
    wrapper
      .find('li.sector')
      .at(0)
      .find('a')
      .simulate('click');
    expect(toggleCategoriesAction).toHaveBeenCalledWith(props.mapId, 'Regions', -1);
    expect(toggleCategoriesAction).toHaveBeenCalledWith(props.mapId, 'Regions', -1);
    expect(toggleCategoriesAction).toHaveBeenCalledTimes(1);

    // test onRegionClick
    const changeRegionAction = jest.spyOn(Actions, 'changeRegion');
    wrapper.find('#reg1').simulate('change');
    expect(changeRegionAction).toHaveBeenCalledWith('reg1');
    expect(changeRegionAction).toHaveBeenCalledTimes(1);
  });

  it('Renders correctly when search bar is active', () => {
    initialState.APP = {
      ...initialState.APP,
      searchBar: true,
      appColor: 'red',
    };
    const store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <Menu {...props} />
      </Provider>
    );
    expect(wrapper.find('SearchBar').length).toEqual(1);
    expect(wrapper.find('SearchBar').props()).toMatchSnapshot('search bar on');
  });
});
