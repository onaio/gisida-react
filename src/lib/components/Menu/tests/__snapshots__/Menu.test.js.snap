// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component Menu component renders correctly with all props 1`] = `
<Connect(Menu)
  mapId="map-1"
  menuId="menu-1"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Menu
    categories={
      Array [
        Object {
          "category": "Boundaries & Labels",
          "layers": Array [
            Object {
              "Place Labels": Object {
                "category": "Place Labels",
                "layers": Array [
                  Object {
                    "category": "Place Labels",
                    "id": "region-labels",
                    "label": "Region labels",
                  },
                  Object {
                    "category": "Place Labels",
                    "id": "district-labels",
                    "label": "District labels",
                  },
                  Object {
                    "category": "Place Labels",
                    "id": "banadir-labels",
                    "label": "Banadir labels",
                  },
                ],
                "parent": "Boundaries & Labels",
              },
            },
            Object {
              "Boundaries": Object {
                "category": "Boundaries",
                "layers": Array [
                  Object {
                    "category": "Boundaries",
                    "id": "region-boundaries",
                    "label": "Region boundaries",
                  },
                  Object {
                    "category": "Boundaries",
                    "id": "district-boundaries",
                    "label": "Districts boundaries",
                  },
                ],
                "parent": "Boundaries & Labels",
              },
            },
          ],
        },
      ]
    }
    currentRegion="reg2"
    dispatch={[Function]}
    hasNavBar={false}
    loaded={true}
    mapId="map-1"
    mapTargetId=""
    menuId="sector-menu-1"
    openCategories={
      Array [
        "Boundaries",
      ]
    }
    preparedLayers={
      Object {
        "Regionboundaries": Object {
          "category": "Boundaries",
          "filters": Object {},
          "id": "regionboundaries",
          "isChartMin": true,
          "isLoading": false,
          "label": "Regionboundaries",
          "legendBottom": 40,
          "loaded": true,
          "paint": Object {
            "line-color": "#444",
            "line-opacity": 0.6,
            "line-width": 2,
          },
          "source": Object {
            "layer": "som_adm_1-6k13xf",
            "type": "vector",
            "url": "mapbox://ona.bnc2rzsw",
          },
          "styleSpec": Object {
            "id": "regionboundaries",
            "layout": Object {},
            "paint": Object {
              "line-color": "#444",
              "line-opacity": 0.6,
              "line-width": 2,
            },
            "source": Object {
              "type": "vector",
              "url": "mapbox://ona.bnc2rzsw",
            },
            "source-layer": "som_adm_1-6k13xf",
            "type": "line",
            "visible": true,
          },
          "type": "line",
          "visible": true,
        },
      }
    }
    regions={
      Array [
        Object {
          "current": false,
          "name": "reg1",
        },
        Object {
          "current": true,
          "name": "reg2",
        },
      ]
    }
    showMap={false}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div>
      <div
        className="menu-wrapper top"
        id="map-1-menu-wrapper"
        onScroll={[Function]}
        style={
          Object {
            "marginTop": 0,
          }
        }
      >
        <a
          className="open-btn"
          onClick={[Function]}
          style={
            Object {
              "display": "block",
            }
          }
        >
          <span
            className="glyphicon glyphicon-menu-hamburger"
          />
        </a>
        <div
          className="sectors-menu"
          id="map-1-menu"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <a
            className="close-btn"
            onClick={[Function]}
          >
            <span
              className="glyphicon glyphicon-remove"
            />
          </a>
          <ul
            className="sectors"
          >
            <li
              className="sector"
            >
              <a
                onClick={[Function]}
              >
                Regions
                <span
                  className="caret"
                />
              </a>
              <ul
                className="layers"
              >
                <li
                  className="region map-1"
                  key="reg1"
                >
                  <input
                    checked={false}
                    id="reg1"
                    key="reg1"
                    name="region"
                    onChange={[Function]}
                    type="radio"
                    value="reg1"
                  />
                  <label
                    htmlFor="reg1"
                  >
                    reg1
                  </label>
                </li>
                <li
                  className="region map-1"
                  key="reg2"
                >
                  <input
                    checked={true}
                    id="reg2"
                    key="reg2"
                    name="region"
                    onChange={[Function]}
                    type="radio"
                    value="reg2"
                  />
                  <label
                    htmlFor="reg2"
                  >
                    reg2
                  </label>
                </li>
              </ul>
            </li>
            <li />
          </ul>
        </div>
      </div>
    </div>
  </Menu>
</Connect(Menu)>
`;

exports[`Menu Component Menu component renders correctly with all props: Menu props 1`] = `
Object {
  "AUTH": undefined,
  "activeLayerIds": undefined,
  "categories": Array [
    Object {
      "category": "Boundaries & Labels",
      "layers": Array [
        Object {
          "Place Labels": Object {
            "category": "Place Labels",
            "layers": Array [
              Object {
                "category": "Place Labels",
                "id": "region-labels",
                "label": "Region labels",
              },
              Object {
                "category": "Place Labels",
                "id": "district-labels",
                "label": "District labels",
              },
              Object {
                "category": "Place Labels",
                "id": "banadir-labels",
                "label": "Banadir labels",
              },
            ],
            "parent": "Boundaries & Labels",
          },
        },
        Object {
          "Boundaries": Object {
            "category": "Boundaries",
            "layers": Array [
              Object {
                "category": "Boundaries",
                "id": "region-boundaries",
                "label": "Region boundaries",
              },
              Object {
                "category": "Boundaries",
                "id": "district-boundaries",
                "label": "Districts boundaries",
              },
            ],
            "parent": "Boundaries & Labels",
          },
        },
      ],
    },
  ],
  "currentLanguage": undefined,
  "currentRegion": "reg2",
  "dispatch": [Function],
  "hasNavBar": false,
  "highlightText": undefined,
  "hyperLink": undefined,
  "infoText": undefined,
  "languageTranslations": undefined,
  "legendInfoText": undefined,
  "loaded": true,
  "mapId": "map-1",
  "mapTargetId": "",
  "menuId": "sector-menu-1",
  "menuIsOpen": undefined,
  "menuScroll": undefined,
  "noLayerText": undefined,
  "openCategories": Array [
    "Boundaries",
  ],
  "preparedLayers": Object {
    "Regionboundaries": Object {
      "category": "Boundaries",
      "filters": Object {},
      "id": "regionboundaries",
      "isChartMin": true,
      "isLoading": false,
      "label": "Regionboundaries",
      "legendBottom": 40,
      "loaded": true,
      "paint": Object {
        "line-color": "#444",
        "line-opacity": 0.6,
        "line-width": 2,
      },
      "source": Object {
        "layer": "som_adm_1-6k13xf",
        "type": "vector",
        "url": "mapbox://ona.bnc2rzsw",
      },
      "styleSpec": Object {
        "id": "regionboundaries",
        "layout": Object {},
        "paint": Object {
          "line-color": "#444",
          "line-opacity": 0.6,
          "line-width": 2,
        },
        "source": Object {
          "type": "vector",
          "url": "mapbox://ona.bnc2rzsw",
        },
        "source-layer": "som_adm_1-6k13xf",
        "type": "line",
        "visible": true,
      },
      "type": "line",
      "visible": true,
    },
  },
  "regions": Array [
    Object {
      "current": false,
      "name": "reg1",
    },
    Object {
      "current": true,
      "name": "reg2",
    },
  ],
  "showMap": false,
  "showSearchBar": undefined,
  "store": Object {
    "clearActions": [Function],
    "dispatch": [Function],
    "getActions": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "storeSubscription": Subscription {
    "listeners": Object {
      "clear": [Function],
      "get": [Function],
      "notify": [Function],
      "subscribe": [Function],
    },
    "onStateChange": [Function],
    "parentSub": undefined,
    "store": Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    },
    "unsubscribe": [Function],
  },
}
`;

exports[`Menu Component Renders correctly when search bar is active: search bar on 1`] = `
Object {
  "APP": Object {
    "appColor": "red",
    "loaded": true,
    "searchBar": true,
  },
  "LOC": undefined,
  "appColor": "red",
  "dispatch": [Function],
  "handleSearchClick": [Function],
  "handleSearchInput": [Function],
  "mapId": "map-1",
  "openCategoryForLayers": [Function],
  "preparedLayers": Object {
    "Regionboundaries": Object {
      "category": "Boundaries",
      "filters": Object {},
      "id": "regionboundaries",
      "isChartMin": true,
      "isLoading": false,
      "label": "Regionboundaries",
      "legendBottom": 40,
      "loaded": true,
      "paint": Object {
        "line-color": "#444",
        "line-opacity": 0.6,
        "line-width": 2,
      },
      "source": Object {
        "layer": "som_adm_1-6k13xf",
        "type": "vector",
        "url": "mapbox://ona.bnc2rzsw",
      },
      "styleSpec": Object {
        "id": "regionboundaries",
        "layout": Object {},
        "paint": Object {
          "line-color": "#444",
          "line-opacity": 0.6,
          "line-width": 2,
        },
        "source": Object {
          "type": "vector",
          "url": "mapbox://ona.bnc2rzsw",
        },
        "source-layer": "som_adm_1-6k13xf",
        "type": "line",
        "visible": true,
      },
      "type": "line",
      "visible": true,
    },
  },
  "searchBarColor": undefined,
  "searchResultClick": [Function],
  "searching": false,
  "setState": [Function],
  "state": Object {
    "categories": Array [],
    "searchResults": Array [],
    "searching": false,
    "sharedLayers": Array [],
  },
}
`;
